src/V3AstNodes.h

```diff
@@ -7107,6 +7107,46 @@ class AstPowUS : public AstNodeBiop {
     virtual int instrCount() const { return widthInstrs() * instrCountMul() * 10; }
     virtual bool signedFlavor() const { return true; }
 };
+class AstPreAdd : public AstNodeBiop {
+public:
+    AstPreAdd(FileLine* fl, AstNode* lhsp, AstNode* rhsp)
+        : ASTGEN_SUPER(fl, lhsp, rhsp) {
+            dtypeFrom(lhsp);
+        }
+    ASTNODE_NODE_FUNCS(PreAdd)
+    virtual AstNode* cloneType(AstNode* lhsp, AstNode* rhsp) { return new AstPreAdd(this->fileline(), lhsp, rhsp); }
+    virtual void numberOperate(V3Number& out, const V3Number& lhs, const V3Number& rhs) {
+        out.opAdd(lhs, rhs);
+    }
+    virtual string emitVerilog() { return "%k(++%r)"; }
+    virtual string emitC() { return "VL_SUB_%lq(%lW, %P, %li, %ri)"; }
```

Comment:
In answer to your question, EmitC is how we make C code to represent this operator.  Your changes will be removing these before they get to C++ so this would be reasonable:

```suggestion
        virtual string emitC() { V3ERROR_NA_RETURN(""); }
```